sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -118,8 +118,8 @@ private[hive] object HiveShim {
    *
    * @param functionClassName UDF class name
    */
-  private[hive] case class HiveFunctionWrapper(var functionClassName: String)
-    extends java.io.Externalizable {
+  private[hive] case class HiveFunctionWrapper(var functionClassName: String,
+    var instance: AnyRef = null) extends java.io.Externalizable with Logging {
```

Comment:
Also, do you need to add the logging trait here? It looks like it's not used anywhere.